zerver/decorator.py

```diff
@@ -278,13 +305,15 @@ def _wrapped_view_func(request, *args, **kwargs):
 # look it up anyway.  It is deprecated in favor on the REST API
 # versions.
 def authenticated_api_view(view_func):
+    # type: (Callable[..., HttpResponse]) -> Callable[..., HttpResponse]
     @csrf_exempt
     @require_post
     @has_request_variables
     @wraps(view_func)
     def _wrapped_view_func(request, email=REQ(), api_key=REQ('api_key', default=None),
                            api_key_legacy=REQ('api-key', default=None),
                            *args, **kwargs):
+        # type: (HttpRequest, str, Optional[text_type], Optional[text_type], *Any, **Any) -> HttpResponse
```

Comment:
I think email's type should be `text_type`. Do you get an error if you set it to `text_type`?